redhat-developer/web-terminal-tooling
# Improve generating versions at runtime

## Description

Workaround multiple issues in the get-tooling-version.sh script:
* Ignore CLIs that are not installed; some CLIs may not be available in downstream, leaving an empty field (e.g. `rhoas`)
* Fix generating version for `oc` CLI, as `releaseClientVersion` isn't always available

Also update CLIs to latest available version:
* helm 3.9.0 -> 3.10.1
* odo 3.5.0 -> 3.6.0

## How to test
1. Verify that build logs 
    ```
    Installed tools:
    Command    Version         Name
    oc         4.12.0          OpenShift CLI
    kubectl    1.24.1          Kubernetes CLI
    kustomize  4.5.7           Kustomize CLI
    helm       3.10.1          Helm CLI
    kn         1.5.0           KNative CLI
    tkn        0.24.1          Tekton CLI
    subctl     0.14.1          Submariner CLI
    odo        3.6.0           Red Hat OpenShift Developer CLI
    virtctl    0.58.0          KubeVirt CLI
    rhoas      0.52.0          Red Hat OpenShift Application Services CLI
    jq         1.6             jq
    ```
2. Verify that running `help` within the container prints the same information.
